table_foosball/database_test.rb

```diff
@@ -0,0 +1,139 @@
+require_relative './database'
+require 'minitest/autorun'
+
+class DatabaseTest < Minitest::Test
+  class Foo
+    def initialize(*args)
+      i = 0
+      args.each { |param| instance_variable_set("@var#{i+=1}", param) }
+      @timestamp = Time.now
+    end
+
+    def ==(other)
+      instance_variables.each do |variable_name|
+        return false unless instance_variable_get(variable_name) == other.instance_variable_get(variable_name)
+      end
+
+      true
+    end
+
+    alias eql? ==
+
+    def hash
```

Comment:
So I implemented this to be able to compare hashes containing objects with instance variables as their keys/values. My theory is that `Hash#==` compares objects using their hash values (`#hash`), as hashes containing objects with identical variables would come back as not-equal when I wrote these tests. The docs on `Hash#==` tell me that it uses `Object#==`, which I thought I'd overwritten by `Foo#==`.

Am I on the right track here? And how do you (@Sirupsen) search Ruby's source code for method definitions?